setup.py

```diff
@@ -4,34 +4,136 @@
-          'License :: OSI Approved :: GNU General Public License (GPL)',
-          'Operating System :: MacOS',
-          'Operating System :: Microsoft :: Windows',
-          'Operating System :: POSIX',
-          'Operating System :: Unix',
-          'Programming Language :: Python :: 2',
-          'Programming Language :: Python :: 3',
-          'Topic :: Multimedia :: Graphics',
-          ],
-       keywords    = ['freetype', 'font'],
-     )
+
+from setuptools import Extension, setup
+from setuptools.command.build_ext import build_ext
+
+cmdclass = {}
+try:
+    from wheel.bdist_wheel import bdist_wheel
+except ImportError:
+    print("warning: wheel package is not installed", file=sys.stderr)
+else:
+
+    class UniversalBdistWheel(bdist_wheel):
+        def get_tag(self):
+            return (
+                'py2.py3',
+                'none',
+            ) + bdist_wheel.get_tag(self)[2:]
+
+    cmdclass['bdist_wheel'] = UniversalBdistWheel
+
+
+class SharedLibrary(Extension):
+    """Object that describes the library (filename) and how to make it."""
+    if sys.platform == "darwin":
+        suffix = ".dylib"
+    elif sys.platform == "win32":
+        suffix = ".dll"
+    else:
+        suffix = ".so"
+
+    def __init__(self, name, cmd, cwd=".", output_dir=".", env=None):
+        Extension.__init__(self, name, sources=[])
+        self.cmd = cmd
+        self.cwd = path.normpath(cwd)
+        self.output_dir = path.normpath(output_dir)
+        self.env = env or dict(os.environ)
+
+
+class SharedLibBuildExt(build_ext):
+    """Object representing command to produce and install a shared library."""
+
+    # Needed to make setuptools and wheel believe they're looking at an
+    # extension instead of a shared library.
+    def get_ext_filename(self, ext_name):
+        for ext in self.extensions:
+            if isinstance(ext, SharedLibrary):
+                return os.path.join(*ext_name.split('.')) + ext.suffix
+        return build_ext.get_ext_filename(self, ext_name)
+
+    def build_extension(self, ext):
+        if not isinstance(ext, SharedLibrary):
+            build_ext.build_extension(self, ext)
+            return
+
+        distutils.log.info("running '{}'".format(ext.cmd))
+        if not self.dry_run:
+            rv = subprocess.Popen(
+                ext.cmd, cwd=ext.cwd, env=ext.env, shell=True).wait()
+            if rv != 0:
+                sys.exit(rv)
+
+        lib_name = ext.name.split(".")[-1] + ext.suffix
+        lib_fullpath = path.join(ext.output_dir, lib_name)
+        dest_path = self.get_ext_fullpath(ext.name)
+
+        distutils.dir_util.mkpath(
+            path.dirname(dest_path),
+            verbose=self.verbose,
+            dry_run=self.dry_run)
+
+        distutils.file_util.copy_file(
+            lib_fullpath,
+            dest_path,
+            verbose=self.verbose,
+            dry_run=self.dry_run)
+
+
+cmdclass['build_ext'] = SharedLibBuildExt
```

Comment:
You should pass an empty cmdclass dict if FREETYPEPY_BUNDLE_FT is not set. You don’t want to override the default bdist_wheel or build_ext commands in the case you are not compiling the embedded shared library.

Reply:
Should cmdclass include just `cmdclass['bdist_wheel'] = UniversalBdistWheel` even if not bundling FreeType?

Reply:
Correct, leave it empty.